lib/at_cmd/at_cmd.c

```diff
@@ -76,6 +76,25 @@ static int open_socket(void)
 	return 0;
 }
 
+/*
+ * Do any validation of an AT command not performed by the lower layers or
+ * by the modem.
+ */
+static int check_cmd(const char *cmd)
+{
+	if (cmd == NULL) {
+		return -EINVAL;
+	}
+
+	/* Check for the presence one printable non-whitespace character */
+	for (const char *c = cmd; *c != '\0'; c++) {
```

Comment:
This looks a bit unsafe, relying only on 0-termination. Do you really need to check the whole string, isn't enough to only check the first character? And fail the string if begin with a whitespace character. In the at command parser library we have some utility commands to detect different parts of the AT strings. Maybe these functions can be re-used in this context.

This expression will also accept all of the extended ascii characters, which I am not sure are valid AT string characters.